lib/helpers.bash

```diff
@@ -849,3 +849,30 @@ then
     fi
   }
 fi
+
+# `_bash-it-find-in-ancestor` uses the shell's ability to run a function in
+# a subshell to simplify our search to a simple `cd ..` and `[[ -r $1 ]]`
+# without any external dependencies. Let the shell do what it's good at.
+function _bash-it-find-in-ancestor() (
+	about 'searches parents of the current directory for the specified file name'
+	group 'helpers'
+	param '1: name of file or folder to search for'
+	param '2: alternate name to search for'
+	returns '0: prints path to matching parent directory to stdout'
+	returns '1: no match found'
+	returns '2: improper usage of shell builtin' # uncommon
+	example '_bash-it-find-in-ancestor .git'
+	example '_bash-it-find-in-ancestor Makefile'
+
+	local kin
+	while [[ "${PWD}" != '/' ]]; do
```

Comment:
We still might want to mention somewhere (even just a command above the `while` statement), that we purposefully do not check `/` for matches.

You want the reader to know that this behavior is a decision and not an unintended side effect.

Reply:
I suppose I could reintroduce `$common_ancestor` and compare against that instead of `/`? Do you think there's a use case for looking for something in root (do CI systems sometimes put stuff in root? Docker containers?)? This isn't meant to be as full-fledged as a `find` replacement. For simple helpers like this, I specifically value simplicity.

Reply:
I think its fine if it doesn't search `/` but just want it documented in the code (again even just a comment right above the while loop) to show it was an intentional decision ... If we decide later that there's value in actually checking `/` someone can update it at that time.